apps/OpenSpace/ext/launcher/src/profile/horizonsdialog.cpp

```diff
@@ -0,0 +1,1178 @@
+
+            if (!start.isValid() || !end.isValid()) {
+                if (timeRange.first.empty() || timeRange.second.empty()) {
+                    _errorMsg->setText("Could not find time range");
+                    appendLog(
+                        "Could not find time range in Horizons file",
+                        LogLevel::Error
+                    );
+                }
+                else {
+                    _errorMsg->setText("Could not read time range");
+                    appendLog(fmt::format("Could not read time range '{}' to '{}'",
+                        timeRange.first, timeRange.second), LogLevel::Error
+                    );
+                }
+                return std::pair<std::string, std::string>();
+            }
+            else {
+                appendLog(
+                    fmt::format("Could not find all time range information. Latest "
+                        "Horizons mesage: {}", _latestHorizonsError
+                    ),
+                    LogLevel::Warning
+                );
+            }
+        }
+    }
+    return timeRange;
+}
+
+bool HorizonsDialog::handleRequest() {
+    if (!isValidInput()) {
+        return false;
+    }
+
+    // Reset
+    _errorMsg->clear();
+    cleanAllWidgets();
+    _importTimeButton->hide();
+    _validTimeRange = std::pair<std::string, std::string>();
+    _latestHorizonsError.clear();
+
+    std::string url = constructUrl();
+
+    _chooseObserverCombo->clear();
+    _chooseObserverCombo->hide();
+    _chooseTargetCombo->clear();
+    _chooseTargetCombo->hide();
+
+    json answer = sendRequest(url);
+    if (answer.empty()) {
+        _errorMsg->setText("Connection error");
+        return false;
+    }
+
+    openspace::HorizonsFile file = handleAnswer(answer);
+    if (file.isEmpty()) {
+        return false;
+    }
+
+    _horizonsFile = std::move(file);
+    openspace::HorizonsFile::ResultCode result =
+        _horizonsFile.isValidHorizonsFile();
+
+    bool isValid = handleResult(result);
+
+    if (!isValid && std::filesystem::is_regular_file(_horizonsFile.file())) {
+        std::string newName = _horizonsFile.file().filename().string();
+        newName = newName.substr(0, newName.size() - newName.find_first_of('.'));
+
+        std::filesystem::path oldFile = _horizonsFile.file();
+        std::filesystem::path newFile =_horizonsFile.file().replace_filename(
+            newName + "_error.txt"
+        );
+
+        std::filesystem::rename(oldFile, newFile);
+
+        appendLog(
+            fmt::format("For more information, see the saved error file {}",
+            newFile),
+            LogLevel::Info
+        );
+    }
+
+    return isValid;
+}
+
+bool HorizonsDialog::isValidInput() {
+    bool isValid = true;
+    std::string message;
+
+    // File
+    if (_fileEdit->text().isEmpty()) {
+        isValid = false;
+        message = "File path not selected";
+        styleLabel(_fileLabel, true);
+    }
+
```

Comment:
I liked the idea to visually separate the validation of each field and make it look less cluttered, but I can change if if we prefer the other way of course.

Reply:
How about the restructuring the function a bit more to achieve that?
```cpp
bool HorizonsDialog::isValidInput() {
    if (_fileEdit->text().isEmpty()) {
        _errorMsg->setText("File path not selected");
        styleLabel(_fileLabel, true);
        return false;
    }

    // Target field
    if (_targetEdit->text().isEmpty() &&
            (_chooseTargetCombo->count() > 0 &&
             _chooseTargetCombo->currentIndex() == 0))
    {
        _errorMsg->setText("Target not selected");
        styleLabel(_fileLabel, true);
        return false;
    }

    if (_targetEdit->text().isEmpty() && _chooseTargetCombo->count() == 0) {
        _errorMsg->setText("Target not selected");
        styleLabel(_fileLabel, true);
        return false;
    }

   // etc

   return true;
```


This way it is not one `if` statement with a lot of `else if` attachments, but it is one `if` statement after the other so the empty lines don't matter since it is not one conceptual thing anymore